src/server/storage/groups.js

```diff
@@ -48,33 +60,50 @@
         collection = db.collection('groups');
     };
 
-    GroupStore.new = function(name) {
-        logger.trace(`creating new group: ${name}`);
+    // in: groupName and owner's username
+    GroupStore.new = async function(name, owner) {
+        logger.trace(`creating new group: ${owner}/${name}`);
+        let curGroup = await this.findOne(name, owner);
+        if (curGroup) throw new Error(`group ${owner}/${name} exists`);
```

Comment:
this might be better to just be "Group already exists" (if it is going to be shown to the end user).